src/xunit.performance.api/Profilers/Etw/SimpleTraceEventParser.cs

```diff
@@ -0,0 +1,378 @@
+                        process.PerformanceMonitorCounterData.Add(performanceMonitorCounter, 0);
+                    process.PerformanceMonitorCounterData[performanceMonitorCounter] += pmcSamplingIntervals[obj.ProfileSource];
+
+                    // Is the IP in the kernel (Under this process)?
+                    if (defaultNtoskrnlModule != null && obj.IsInTimeAndAddressIntervals(defaultNtoskrnlModule))
+                    {
+                        var krnlModule = process.Modules.SingleOrDefault(m => m.Checksum == defaultNtoskrnlModule.Checksum
+                            && m.AddressSpace == defaultNtoskrnlModule.AddressSpace
+                            && m.FullName == defaultNtoskrnlModule.FullName);
+                        if (krnlModule == null)
+                        {
+                            krnlModule = defaultNtoskrnlModule.Copy();
+                            process.Modules.Add(krnlModule);
+                        }
+                    }
+
+                    var modules = process.Modules
+                        .Where(m => {
+                            return obj.IsInTimeAndAddressIntervals(m);
+                        })
+                        .Select(m => m);
+
+                    if (modules.Count() == 0)
+                    {
+                        // This might fall in managed code. We need to buffer and test it afterwards.
+                        pmcRollovers.Add(new PmcRollover {
+                            InstructionPointer = obj.InstructionPointer,
+                            ProcessId = obj.ProcessID,
+                            ProfileSourceId = obj.ProfileSource,
+                            TimeStamp = obj.TimeStamp,
+                            SamplingInterval = pmcSamplingIntervals[obj.ProfileSource],
+                        });
+                        return;
+                    }
+
+                    modules.ForEach(module => {
+                        if (!module.PerformanceMonitorCounterData.ContainsKey(performanceMonitorCounter))
+                            module.PerformanceMonitorCounterData.Add(performanceMonitorCounter, 0);
+                        module.PerformanceMonitorCounterData[performanceMonitorCounter] += pmcSamplingIntervals[obj.ProfileSource];
+                    });
+                };
+
+                ////////////////////////////////////////////////////////////////
+                // .NET modules
+                parser.Source.Clr.LoaderModuleLoad += (ModuleLoadUnloadTraceData obj) => {
+                    var process = processes.SingleOrDefault(p => p.Id == obj.ProcessID);
+                    if (process == null)
+                        return;
+
+                    // FIXME: Is this condition correct? Will a module be always loaded, on ImageLoad, before getting here?
+                    var module = process.Modules
+                        .SingleOrDefault(m => m.FullName == obj.ModuleILPath);
```

Comment:
I think that this is fine on core but would be broken on desktop.  The ModuleILPath and the ModuleNativePath are equivalent on core, but they are different on desktop.  I believe that you will only ever actually use one of them.  The native image contains the IL so even if a method isn't pre-compiled, the IL can be pulled from the native image.  You may wish to see if the ModuleNativePath is specified and use that first before the ModuleILPath.  I believe that ModuleNativePath is only specified if the we were able to bind to it.

Reply:
Yes, I did noticed that `obj.ModuleNativePath` is an empty string most of the times.

---
In reply to: [146057410](https://github.com/Microsoft/xunit-performance/pull/242#discussion_r146057410) [](ancestors = 146057410)

Reply:
Do you mean something like this:
`var modulePath = string.IsNullOrEmpty(obj.ModuleNativePath) ? obj.ModuleILPath : obj.ModuleNativePath;`

---
In reply to: [146083680](https://github.com/Microsoft/xunit-performance/pull/242#discussion_r146083680) [](ancestors = 146083680,146057410)